src/oncall/api/v0/event_link.py

```diff
@@ -59,4 +61,84 @@ def on_delete(req, resp, link_id):
 
     connection.commit()
     cursor.close()
-    connection.close()
\ No newline at end of file
+    connection.close()
+
+
+@login_required
+def on_put(req, resp, link_id):
+    """
+    Update an event by link_id; anyone can update any event within the team.
+    Only username can be updated using this endpoint.
+
+    **Example request:**
+
+    .. sourcecode:: http
+
+        PUT /api/v0/events/link/1234 HTTP/1.1
+        Content-Type: application/json
+
+        {
+            "user": "asmith",
+        }
+
+    :statuscode 200: Successful update
+
+    """
+    data = load_json_body(req)
+    user = data.get('user')
+    if user is None:
+        raise HTTPBadRequest('Bad request for linked event update', 'Missing user param')
+    connection = db.connect()
+    cursor = connection.cursor(db.DictCursor)
+
+    try:
+        cursor.execute('''SELECT
+                        `event`.`start`,
+                        `event`.`end`,
+                        `event`.`user_id`,
+                        `event`.`role_id`,
+                        `event`.`id`,
+                        `team`.`name` AS `team`,
+                        `role`.`name` AS `role`,
+                        `user`.`name` AS `user`,
+                        `user`.`full_name`,
+                        `event`.`team_id`
+                    FROM `event`
+                    JOIN `team` ON `event`.`team_id` = `team`.`id`
+                    JOIN `role` ON `event`.`role_id` = `role`.`id`
+                    JOIN `user` ON `event`.`user_id` = `user`.`id`
+                    WHERE `event`.`link_id`=%s''', link_id)
+        event_data = cursor.fetchall()
+        if len(event_data) == 0:
+            raise HTTPNotFound()
+        event_summary = event_data[0].copy()
+        event_summary['end'] = max(event_data, key=itemgetter('end'))['end']
+        event_summary['start'] = min(event_data, key=itemgetter('start'))['start']
+        if not user_in_team_by_name(cursor, user, event_summary['team']):
+            raise HTTPBadRequest('Invalid event update', 'Event user must be part of the team')
+
+        now = time.time()
+        if event_summary['end'] < now:
+            raise HTTPBadRequest('Invalid event update',
+                                 'Editing events in the past not allowed')
+        check_calendar_auth(event_summary['team'], req)
+
+        cursor.execute('SELECT `id` FROM `user` WHERE `name` = %s', user)
+        user_id = cursor.fetchone()['id']
```

Comment:
this will raise an exception if fetchone() returns None. You should probably catch that to avoid a 5xx.